src/main/java/no/finntech/lambdacompanion/Success.java

```diff
@@ -32,6 +34,16 @@ public Success(T t) {
     }
 
     @Override
+    public Try<T> filter(final Predicate<T> predicate) {
+        Objects.requireNonNull(predicate);
+        if (predicate.test(t)) {
+            return this;
+        } else {
+            return new Failure<>(new RuntimeException());
```

Comment:
A bit unsure about leaving it as a raw `RuntimeException` without even a message as it makes it impossible to know what happened.
What about a `NoSuchElementException("No value present")` just like in java8's `Optional#get` class? Maybe even with a more self explanatory message like f.ex. `"No success present after filtering"` or something like that?